### PR TITLE
NameResolution-Tests: enable GetHostEntryAsync_InvalidHost_LogsError on Windows

### LINKED ISSUE #47104: Test failed: System.Net.NameResolution.Tests.LoggingTest.GetHostEntryAsync_InvalidHost_LogsError
Job: [runtime-coreclr libraries-jitstress 20210116.1](https://dev.azure.com/dnceng/public/_build/results?buildId=954677&view=ms.vss-test-web.build-test-results-tab&runId=30122164&resultId=151744&paneView=debug)

Failed tests:
```
net6.0-Linux-Release-arm-CoreCLR_checked-zapdisable-(Ubuntu.1804.Arm32.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7-bfcd90a-20200121150440
net6.0-Linux-Release-arm64-CoreCLR_checked-zapdisable-(Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-a45aeeb-20190620155855
        - System.Net.NameResolution.Tests.LoggingTest.GetHostEntryAsync_InvalidHost_LogsError
```

Error message:
```
events.Count should be > 0
Expected: True
Actual:   False

stack trace:
at System.Net.NameResolution.Tests.LoggingTest.GetHostEntryAsync_InvalidHost_LogsError() in /_/src/libraries/System.Net.NameResolution/tests/FunctionalTests/LoggingTest.cs:line 91
--- End of stack trace from previous location ---
```

### PR DESCRIPTION
**Pull request by gfoidl**
In https://github.com/dotnet/runtime/pull/34633#issuecomment-746681771 a product bug in regards of logging was found.
This didn't show up in tests, as it was disabled for Windows https://github.com/dotnet/runtime/blob/911b1d395ef4e59c63048003d676e295414c8d15/src/libraries/System.Net.NameResolution/tests/FunctionalTests/LoggingTest.cs#L66 and prior #34633 Unix was async-over-sync, so didn't touch this test in the same way is it does now.

On local testing I couldn't spot any unrealiable behavior, per https://github.com/dotnet/runtime/pull/34633#discussion_r544521828 we should give this a try to see what happens.

---

Edit: Fixes https://github.com/dotnet/runtime/issues/47104

### CODE REVIEW

File: src/libraries/System.Net.NameResolution/tests/FunctionalTests/LoggingTest.cs [hunk 1]
```diff
@@ -63,7 +63,6 @@ public void GetHostEntry_InvalidHost_LogsError()
         }
 
         [ConditionalFact]
-        [PlatformSpecific(~TestPlatforms.Windows)]  // Unreliable on Windows.
```
Comment (stephentoub) [inline src/libraries/System.Net.NameResolution/tests/FunctionalTests/LoggingTest.cs:66]: I'm skeptical any time I see a construct like:
```C#
await Task.Delay(100).ConfigureAwait(false);
```
used for synchronization.  But if it's not been failing on Unix, I'm not sure why it'd be any more likely to fail on Windows.

Comment (gfoidl) [reply src/libraries/System.Net.NameResolution/tests/FunctionalTests/LoggingTest.cs:66]: Yep, that's not ideal. Is for the event source a better way to "wait" until the log is written?

Comment (wfurt) [reply src/libraries/System.Net.NameResolution/tests/FunctionalTests/LoggingTest.cs:66]: I'm wondering if we could emit some other known event and wait for that to arrive. And assume the events are processed in sequence. @MihaZupan  would probably know how the events are processed.

Comment (MihaZupan) [reply src/libraries/System.Net.NameResolution/tests/FunctionalTests/LoggingTest.cs:66]: We should use something like [`WaitForEventAsync`](https://github.com/dotnet/runtime/blob/a3b37a289b00c748eb92bd4eb4dc78146bd72176/src/libraries/System.Net.Sockets/tests/FunctionalTests/TelemetryTest.cs#L418) we used in Sockets telemetry across all such tests.

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**